conans/client/conf/detect.py

```diff
@@ -38,11 +38,19 @@ def _gcc_compiler(output, compiler_exe="gcc"):
         if ret != 0:
             return None
         compiler = "gcc"
+
         installed_version = re.search("([0-9](\.[0-9])?)", out).group()
         # Since GCC 7.1, -dumpversion return the major version number
         # only ("7"). We must use -dumpfullversion to get the full version
         # number ("7.1.1").
         if installed_version:
+            if not "." in installed_version and int(installed_version) >= 7:
+                ret, out = _execute('%s -dumpfullversion' % compiler_exe)
+                if ret != 0:
+                    return None
+
+            installed_version = re.search("([0-9](\.[0-9])?)", out).group()
```

Comment:
What is that about? Specific for AIX? different output there or..? If it is not related please remove it and open a new PR with the "refactors".

Reply:
Ahh ... thats related to an other problem. I will have a look if there is already a bug filed and move it to a separate PR.

```
~$ gcc -dumpversion
7
~$ gcc -dumpfullversion
7.3.0
```

gcc version detection is not working anymore correctly with newer gcc.

Not sure how fast I can update the PR as I am on a tight schedule at the moment. Hopefully I can resolve it soon.

Reply:
I have gcc 7 and the detection is working just fine. For gcc >= 5 we are taking only the major version, as they changed the versioning scheme also.